homeassistant/components/asuswrt/config_flow.py

```diff
@@ -76,25 +75,28 @@ def _show_setup_form(self, user_input=None, errors=None):
         if user_input is None:
             user_input = {}
 
+        schema = {
+            vol.Required(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
+            vol.Required(CONF_USERNAME, default=user_input.get(CONF_USERNAME, "")): str,
+            vol.Optional(CONF_PASSWORD)
+            if self.show_advanced_options
+            else vol.Required(CONF_PASSWORD): str,
+            vol.Required(CONF_PROTOCOL, default=PROTOCOL_SSH): vol.In(
+                {PROTOCOL_SSH: "SSH", PROTOCOL_TELNET: "Telnet"}
+            ),
+        }
+
+        if self.show_advanced_options:
+            schema[vol.Optional(CONF_PORT)] = cv.port
+            schema[vol.Optional(CONF_SSH_KEY)] = str
```

Comment:
We could also ask this in a 2nd step if `PROTOCOL_SSH` was selected

Reply:
Port can be changed also for telnet. For the ssh key ok, but just in case the password is empty. 
But I'm working to an alternative connection method, so I will come back to this when ready.